static/app/views/routeError.tsx

```diff
@@ -7,51 +7,53 @@ import Alert from 'sentry/components/alert';
 import List from 'sentry/components/list';
 import ListItem from 'sentry/components/list/listItem';
 import {t, tct} from 'sentry/locale';
+import OrganizationStore from 'sentry/stores/organizationStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
 import space from 'sentry/styles/space';
-import {Organization, Project} from 'sentry/types';
+import {Project} from 'sentry/types';
 import getRouteStringFromRoutes from 'sentry/utils/getRouteStringFromRoutes';
-import withOrganization from 'sentry/utils/withOrganization';
 import withProject from 'sentry/utils/withProject';
 
 type Props = WithRouterProps & {
-  organization: Organization;
   /**
    * Disable logging to Sentry
    */
   disableLogSentry?: boolean;
+  /**
+   * Disable the report dialog
+   */
   disableReport?: boolean;
   error?: Error;
   project?: Project;
 };
 
-class RouteError extends Component<Props> {
-  UNSAFE_componentWillMount() {
-    const {error} = this.props;
-    const {disableLogSentry, disableReport, organization, project, routes} = this.props;
+function RouteError({error, disableLogSentry, disableReport, project, routes}: Props) {
+  const {organization} = useLegacyStore(OrganizationStore);
 
+  useEffect(() => {
     if (disableLogSentry) {
-      return;
+      return undefined;
     }
     if (!error) {
-      return;
+      return undefined;
     }
 
     const route = getRouteStringFromRoutes(routes);
-    const enrichScopeContext = scope => {
+    const enrichScopeContext = (scope: Sentry.Scope) => {
       scope.setExtra('route', route);
-      scope.setExtra('orgFeatures', (organization && organization.features) || []);
-      scope.setExtra('orgAccess', (organization && organization.access) || []);
-      scope.setExtra('projectFeatures', (project && project.features) || []);
+      scope.setExtra('orgFeatures', organization?.features ?? []);
+      scope.setExtra('orgAccess', organization?.access ?? []);
+      scope.setExtra('projectFeatures', project?.features ?? []);
       return scope;
     };
 
     if (route) {
-      /**
-       * Unexpectedly, error.message would sometimes not have a setter property, causing another exception to be thrown,
-       * and losing the original error in the process. Wrapping the mutation in a try-catch in an attempt to preserve
-       * the original error for logging.
-       * See https://github.com/getsentry/sentry/issues/16314 for more details.
-       */
+      // Unexpectedly, error.message would sometimes not have a setter
+      // property, causing another exception to be thrown, and losing the
+      // original error in the process. Wrapping the mutation in a try-catch in
+      // an attempt to preserve the original error for logging.
+      //
```

Comment:
```suggestion
```

Reply:
This line is to start a new paragraph :)